### PR TITLE
Closes #94 tests for assertionsr@devel

### LINKED ISSUE #94: General Issue: Increase test coverage for functions in `assertions.R`
### Background Information

We need to increase test coverage for our functions. The below screenshot has the function and gap in test coverage that we need to write additional unit tests

Most of these gaps surround the messages a user receives when something goes awry.

image

### Definition of Done

Unit tests are written that increase the coverage of the `assertions.R` file

### PR DESCRIPTION
**Pull request by AniaGolab**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by AniaGolab** [comment]
HI @bms63 and @thomas-neitmann, I added 57 tests on **assertions.r** with overall 100% test coverage. I also added a description to the _values_ of the _assert_character_scalar_  and in the _assert_character_scalar_ itself in the **assertions.r**. Please let me know if you want me to add this change into the NEWS function too? Or if you have any other feedback. Thank you :)

**Comment by bms63** [comment]
Wow.  You have single-handedly increased the test coverage from 50% to 90%!  A true admiral legend

File: tests/testthat/test-assertions.R [hunk 1]
```diff
@@ -138,26 +141,754 @@ test_that("assert_vars Test 11: error if unexpected input", {
   expect_error(assert_vars(vars(APERSDT = APxxSDT, APxxEDT), expect_names = TRUE))
 })
 
+# assert_data_frame ----
+test_that("Test 12 : `assert_data_frame` doesn't throw an error
```
Comment (bms63) [inline tests/testthat/test-assertions.R:145]:
```suggestion
test_that("Test 12 : `assert_data_frame` does not throw an error
```

File: tests/testthat/test-assertions.R [hunk 2]
```diff
@@ -138,26 +141,754 @@ test_that("assert_vars Test 11: error if unexpected input", {
...
+    assert_character_scalar(character, optional = TRUE)
+  }
+
+  expect_invisible(
+    example_fun(NULL)
+  )
+})
+
+test_that("Test 16 : `assert_character_scalar` doesn't throw an error if case_sensitive is FALSE", {
+  example_fun <- function(character) {
```
Comment (bms63) [inline tests/testthat/test-assertions.R:193]:
```suggestion
test_that("Test 16 : `assert_character_scalar` does not throw an error if case_sensitive is FALSE", {
  example_fun <- function(character) {
```

File: tests/testthat/test-assertions.R [hunk 3]
```diff
@@ -138,26 +141,754 @@ test_that("assert_vars Test 11: error if unexpected input", {
...
+test_that("Test 18 : `assert_character_vector` throws an error if `arg` not a character vector", {
+  arg <- c(1, 2, 3)
+
+  expect_error(
+    assert_character_vector(arg)
+  )
+})
+
+test_that("Test 19 : `assert_character_vector` throws an error
+          if values are not NULL and `arg` not in values", {
```
Comment (bms63) [inline tests/testthat/test-assertions.R:222]:
```suggestion
          if values are not NULL and `arg` is not in values", {
```

File: tests/testthat/test-assertions.R [hunk 4]
```diff
@@ -138,26 +141,754 @@ test_that("assert_vars Test 11: error if unexpected input", {
...
+    assert_logical_scalar(arg)
+  }
+  arg <- c()
+  expect_error(example_fun(NA))
+  expect_error(example_fun(arg))
+  expect_error(example_fun("test"))
+})
+
+# assert_symbol ----
+test_that("Test 22 : `assert_symbol` doesn't throw an error if optional = TRUE and `arg` = NULL", {
```
Comment (bms63) [inline tests/testthat/test-assertions.R:255]:
```suggestion
test_that("Test 22 : `assert_symbol` does not throw an error if optional = TRUE and `arg` = NULL", {
```

**Review by bms63 — COMMENTED** [review]
Can you replace `doesn't` with `does not` please?

**Review by bms63 — COMMENTED** [review]
Just a few minor grammar issues to address.  Does this give 100% coverage for all the `assertions.R` functions?

**Review by bms63 — CHANGES_REQUESTED** [review]
Can you remove this from the file and directly call `admiral.test::admiral_dm`

image

**Comment by AniaGolab** [comment]
Thank you for the comments Ben! I applied the changes, please let me know if there is anything else to edit :)

**Review by bms63 — APPROVED** [review]
Many thanks for this @AniaGolab.  You single-handedly brought us from 40% to 90% test coverage!!

**Pull request merged**